### PR TITLE
[GC] Update br_on_cast: the text format also no longer has a heap type

### PR DESCRIPTION
**Pull request by kripken**
I removed it from the binary format earlier, but forgot the text.

As a result, we cannot handle a `br_on_cast` with an unreachable RTT. The
binary format solves the problem by ignoring unreachable code, and this makes
the text format do the same.

A nice benefit of this is that we can remove the `castType` extra field.

### CODE REVIEW

File: scripts/test/shared.py [hunk 1]
```diff
@@ -493,18 +493,8 @@ def minify_check(wast, verify_final_result=True):
...
-    assert os.path.exists('b.wast')
-    if verify_final_result:
-        expected = open('a.wast').read()
-        actual = open('b.wast').read()
-        if actual != expected:
-            fail(actual, expected)
-    if os.path.exists('a.wast'):
-        os.unlink('a.wast')
-    if os.path.exists('b.wast'):
-        os.unlink('b.wast')
```
Comment (tlively) [inline scripts/test/shared.py:507]: Why this change?

**Review by tlively — APPROVED** [review]

File: scripts/test/shared.py (hunk 1 above)
Comment (kripken) [reply scripts/test/shared.py:507]: Sorry, I should have left a comment. See https://github.com/WebAssembly/binaryen/pull/3524 - I added it here too to show tests pass.
